src/os/bluestore/BlueStore.cc

```diff
@@ -4283,9 +4283,11 @@ int BlueStore::_balance_bluefs_freespace(PExtentVector *extents)
     int r = alloc->reserve(gift);
     assert(r == 0);
 
+    thread_local uint64_t hint = 0;
```

Comment:
`static std::atomic_uint64_t`? Nevertheless, the impact of fragmentation is an open question.